tests/spec/core/linter-rules/check-period-in-p.js

```diff
@@ -0,0 +1,60 @@
+fdescribe("Core Linter Rule - 'check-period-in-p'", () => {
+  const ruleName = "check-period-in-p";
+  const config = { lint: { [ruleName]: true } };
+  let rule;
+  beforeAll(async () => {
+    rule = await new Promise(resolve => {
+      require([`core/linter-rules/${ruleName}`], ({ rule }) => resolve(rule));
+    });
+  });
+  it("checks p ending without a period", async () => {
+    const doc = document.implementation.createHTMLDocument("test doc");
+    doc.body.innerHTML = `
+        <section>
+          <p id="no-fullstop">Should pass</p>
+          <p>Should fail.</p>
+          <p id="no-fullstop-at-end">Should. pass</p>
+          <p>Should. Fail.</p>
+          <p id="unnecessary-tests-after-fullstop">Should Pass. Tests</p>
+        </section>
+    `;
+
+    const noFullStop = doc.getElementById("no-fullstop");
+    const noFullStopAtEnd = doc.getElementById("no-fullstop-at-end");
+    const unnecessaryTestsAfterFullStop = doc.getElementById("unnecessary-tests-after-fullstop");
+    
+    const results = await rule.lint(config, doc);
+    const [result] = results;
+    
+    expect(result.offendingElements.length).toEqual(2);
```

Comment:
Make sure that offending elements contain the “respec-offending-element” class. It should have been added automatically, but we want to assure it was.